users/forms.py

```diff
@@ -52,6 +55,16 @@ class Meta:
         model = CustomUser
         fields = "__all__"
 
+    def clean(self) -> Dict[str, Any]:
+        cleaned_data = super().clean()
+        cleaned_is_active = cleaned_data.get("is_active")
+        if self.instance.is_active and not cleaned_is_active:
+            cleaned_data["user_type"] = CustomUser.UserType.EMPLOYEE.name
+            cleaned_data["is_staff"] = False
+            cleaned_data["is_superuser"] = False
+
```

Comment:
This is really only a formality which I added, since it was mentioned that "a deactivated user should lose all of his privileges". However, I should point out that _is_active_ takes priority over _is_staff_ or _is_superuser_. Therefore, even if he does have those privileges, he still won't be able to sign in to the admin panel, as long as he's marked as inactive.
It's because of this irrelevancy that I only tested it manually and did not implement any formal unit test for this particular part. The more important user type change however is tested as a part of the signal's unit test.